index.js

```diff
@@ -57,7 +58,61 @@ module.exports = {
     return new MergeTrees(addonTrees);
   },
 
-  _moduleUnificationTrees() {
+  // Trigger exception if the result of `addon.resolverConfig` method has an unexpected format
+  // Show a warning if there are collisions between addon types or addon collections
+  validateAddonsConfig: function(addonsConfig) {
+
+    let types = {};
+    let collections = {};
+
+    Object.keys(addonsConfig).forEach(addonName => {
+
+      let addonConfig = addonsConfig[addonName];
+      if (addonConfig) {
+        if (typeof(addonConfig) !== 'object') {
+          throw new Error(`"addon.resolverConfig" returns an unexpected value. Addon: ${addonName}.`);
```

Comment:
should we be using `assert`? or is that more an ember thing, rather than a CLI thing?

Reply:
I think it is fine to throw an exception if an addon is wrongly using the `resolverConfig` hook.